tests/chainerx_tests/unit_tests/routines_tests/test_linalg.py

```diff
@@ -351,3 +351,97 @@ def forward_xp(self, inputs, xp):
         a, = inputs
         out = xp.linalg.pinv(a)
         return out,
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    chainer.testing.product({
+        'shape': [(1, 1), (3, 3), (6, 6)],
+        'in_dtypes': ['float32', 'float64']
+    })
+))
+class TestCholesky(NumpyLinalgOpTest):
+
+    def generate_inputs(self):
+        a = numpy.random.random(self.shape).astype(self.in_dtypes)
+        # Make random square matrix a symmetric positive definite one
+        a = numpy.array(a.T.dot(a)) + 1e-6 * numpy.eye(*self.shape)
+        return a,
+
+    def forward_xp(self, inputs, xp):
+        a, = inputs
+
+        # Input has to be symmetrized for backward test to work
+        def symmetrize(A):
+            L = xp.tril(A)
+            return (L + L.T)/2.
```

Comment:
It's only the diagonal that should be halved?

Reply:
Well, it doesn't matter much here `L + L.T` is the important operation here for getting the symmetric matrix.
Usually `(A + A.T)/2` is used for symmetrizing, I took using `tril` from [autograd tests](https://github.com/HIPS/autograd/blob/master/tests/test_linalg.py#L17).

Reply:
Thank you for the explanation.